control-plane/p4RuntimeSerializer.cpp

```diff
@@ -489,7 +489,7 @@ class P4RuntimeSymbolTable : public P4RuntimeSymbolTableIface {
             // resolve hash collisions, the id that we select depends on the order in
             // which the names are hashed. This is why we sort the names above.
             boost::optional<p4rt_id_t> id = probeForId(nameId, [=](uint32_t nameId) {
-                return (resourceType << 24) | (nameId & 0xffffff);
+                return (resourceType << 24) | (nameId & 0xffff);
```

Comment:
AFAIK, there is no reason to revert back to 16-bit of hash value. Are you sure it is not just about updating reference outputs for your internal tests? If there is something making use of the 2nd most significant byte in your internal compiler, you may want to try and figure out if this is strictly required or you can update that code.

Reply:
Using the full 24 bits causes tests to fail with 
```
Traceback (most recent call last):
  File "/mnt/p4c/extensions/p4_tests/p4testutils/p4runtime_base_tests.py", line 852, in handle
    return f(*args, **kwargs)
  File "/mnt/p4c/extensions/p4_tests/p4_16/tor.ptf/test.py", line 71, in runTest
    self.runCommonTest("write_lpm_entries.pb.txt")
  File "/mnt/p4c/extensions/p4_tests/p4_16/tor.ptf/test.py", line 22, in runCommonTest
    response = self.write_request(p4runtime_request)
  File "/mnt/p4c/extensions/p4_tests/p4testutils/p4runtime_base_tests.py", line 514, in write_reques
t
    rep = self._write(req)
  File "/mnt/p4c/extensions/p4_tests/p4testutils/p4runtime_base_tests.py", line 511, in _write
    raise P4RuntimeWriteException(e)
P4RuntimeWriteException: <unprintable P4RuntimeWriteException object>
```
restricting it to 16 bits works fine

Reply:
That's an easy one. IIRC this test hardcodes ids in the "write_lpm_entries.pb.txt" file. So you just need to edit that file manually to include the new ids (with the 24-bit hash value) and it will pass.

Reply:
That is extremely helpful -- thank you.  Hopefully that will make this issue go away.